### DISCUSSION TITLE
How can I keep the current selection and move the cursor?

### DISCUSSION THREAD

**Discussion by zhaob1n** [helix-editor/helix#2776] [Q&A]
I wanna keep the current selection and move the cursor to make other selection. But I don't know how.

**Comment by joegm** [comment]
You can duplicate the selection with `C`, but making a movement will always overwrite the last selection in normal mode. When would you want to freeze a selection as opposed to acting on a selection and then moving on to do something else?

**Reply by zhaob1n** [reply]
for example, I wanna copy or yank several lines as a whole.

**Reply by joegm** [reply]
There are many ways of copying several lines, could you give me a specific example where you'd use this?

**Reply by marc-thieme** [reply]
Chiming in since I tried to do the same. For instance, when I have a paragraph like:
`In section 2.3, you can see XY. As the author notes in ..., this is very important. Also, don't forget to refer to section 4.3. as a reference`

And I would like to split it at each sentence.   
The problem here is discerning the periods from the dots in numbers or ...., which I have to do manually in some way.
I started by selecting each dot in the paragraph and then removing those selections one by one which did not end a sentence. However, since the paragraph was long, I had to focues very hard not to make a mistake and eventually gave up because it would reset my selection each time I pressed a wrong button.

My only other idea would be to `f.` each dot and whenever its a proper period, I press enter for the newline. 

In my opinion, it would be nicer if I could persist my selections. What I mean by that, for instance, is to `f.` through the dots and whenever a dot is proper, I would press a key/execute an `add selection to persistent selections` command. In such a way, I could accumulate my selections without fear of messing up.

**Reply by pascalkuthe** [reply]
`mipS. <ret>` should do the trick

**Reply by marc-thieme** [reply]
Easy enough haha, thanks. 
Still I would like to make an argument for the selection topic.

Imagine a scenario where I need to copy some text from an Email but censor some of it:

`Harry Huffman approves your changes. Please call him under 0374 3614 4748 and discuss further proceedings...`

--> `Harry XXX approves your changes. Please call him under XXX and discuss further proceedings...`
/ --> `Harry ### approves your changes. Please call him under ### and discuss further proceedings...`

and maybe I want to try out the looks of different variants all at once.

In the end, it boils down to wanting to select syntactically arbitrary elements in this paragraph.

The equivalent in mouse editors would be something like ctrl+mouse select to add a selection, I suppose. But is there something equivalent in Helix?

**Reply by pascalkuthe** [reply]
You can keep removing text with A-K. For the most part helix is focused around multi selections instead of incrementally selecting something. Altough what you desctibe is somewhat somewhat similar to #3720

**Reply by marc-thieme** [reply]
Very nice to know, thank you. While reading through the linked issues, I stumbled upon a different behavior which seems odd to me and I would like to ask if it is intented/not doing it another way is deliberate. I just wanted to start a new discussion for that, but I thought that this question will most likely be redundant, concerning I am just starting to familiarize myself with the community, therefore I will just paste the formulation below:

### **When in visual (extend) mode, c-o and c-i should extend the current selection with the selection from the jumplist instead of replacing it?**

### Current State
c-o and c-i are the commands to jump through the jump list. Each entry in the jump list represents a selection. When I press c-o or c-i, it removes the current selection and replaces it with the one from the jumplist. This behavior does not change whether in normal or in visual mode. 

### Problem
As far as I can see, it is impossible to extend a selection with arbitrary text. The discussion about the marks implementation, it seems like, has stalled #703 and #2784 might be similar but different.
Also, it was unintuitive for me that my selection was being replaced in visual mode instead of extended and mimicking the behaviour from normal mode seems redundant to me when the alternative would be intuitive.

### Solution
If I am correct, the arbitrary selection could be easily implemented if c-o and c-i, while in visual mode, would add the previous/next selection from the jump list to the current selection, instead of overriding them. If I wanted to select the confidential data in the previous example 
`Harry Huffman approves your changes. Please call him under 0374 3614 4748 and discuss further proceedings...`, 
I would first select `Huffman` and add it to the jump list (c-s), then select the phone number and c-s it and continue through like that through the file. At the end, I would enter visual mode and walk my jumplist backward (c-o), accumulating a bigger selection with each previously added selection of confidential text.

**Reply by pascalkuthe** [reply]
That's what narks are somewhat for. The jumplist is jot intended for that kind of usecase and not suited for extending the selection because the jumplist can span multiple files. Amy command that can jump to a different file never extends the selection in helix
